_posts/2024-04-29-wait-you-guys-are-writing-unit-tests/wait-you-guys-are-writing-unit-tests.Rmd

```diff
@@ -0,0 +1,95 @@
+---
+title: "Wait, you guys are writing unit tests?"
+description: |
+  A reminder.
+author:
+  - name: Chloe Pou-Prom
+    url: {}
+date: 2024-04-29
+output:
+  distill::distill_article:
+    self_contained: false
+categories:
+    - post-miscellaneous
+draft: true
+preview: unit_tests.png
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = FALSE)
+```
+
+![](unit_tests.png)
+
+Confession: I rarely write unit tests. 😅
+
+Yes, I know. They're important and they help find critical failures. Just enter "Are unit tests worth it?" in your favorite search engine, and you'll end up with a mountain ⛰️ of blog posts, StackOverflow replies, Reddit comments, heated discussions, etc.
+
+Overwhelmingly, the consensus is that *yes, unit tests are worth it*.
+
+And yet... a Very Official Survey[^1] shows that only 55% of data scientists write unit tests.
+
+[^1]: I asked chatgpt to make up a number. Maybe I'll poll our Slack channel to make this number more/less accurate.
```

Comment:
I like the idea of a poll/sharing more about what happens in our team specifically. One poll that might be interesting (if possible to include, or maybe something to consider in a future post) is what people write/wrote unit tests for, whether they were maintained, whether they were useful at finding some issue (and maybe the why behind those questions).

Reply:
Definitely worth having as a future blog post! I very loosely summarized some of this in the concluding section.